src/gui/deletionconfirmationdialog.h

```diff
@@ -30,72 +30,28 @@
 #define DELETIONCONFIRMATIONDIALOG_H
 
 #include <QDialog>
-#include <QPushButton>
 
-#include "base/preferences.h"
 #include "ui_deletionconfirmationdialog.h"
-#include "uithememanager.h"
-#include "utils.h"
+
+namespace Ui
+{
+    class DeletionConfirmationDialog;
+}
 
 class DeletionConfirmationDialog : public QDialog
 {
     Q_OBJECT
 
 public:
-    DeletionConfirmationDialog(QWidget *parent, const int &size, const QString &name, bool defaultDeleteFiles)
-        : QDialog(parent)
-        , m_ui(new Ui::DeletionConfirmationDialog)
-    {
-        m_ui->setupUi(this);
-        if (size == 1)
-            m_ui->label->setText(tr("Are you sure you want to delete '%1' from the transfer list?", "Are you sure you want to delete 'ubuntu-linux-iso' from the transfer list?").arg(name.toHtmlEscaped()));
-        else
-            m_ui->label->setText(tr("Are you sure you want to delete these %1 torrents from the transfer list?", "Are you sure you want to delete these 5 torrents from the transfer list?").arg(QString::number(size)));
-        // Icons
-        const QSize iconSize = Utils::Gui::largeIconSize();
-        m_ui->labelWarning->setPixmap(UIThemeManager::instance()->getIcon("dialog-warning").pixmap(iconSize));
-        m_ui->labelWarning->setFixedWidth(iconSize.width());
-        m_ui->rememberBtn->setIcon(UIThemeManager::instance()->getIcon("object-locked"));
-        m_ui->rememberBtn->setIconSize(Utils::Gui::mediumIconSize());
-
-        m_ui->checkPermDelete->setChecked(defaultDeleteFiles || Preferences::instance()->deleteTorrentFilesAsDefault());
-        connect(m_ui->checkPermDelete, &QCheckBox::clicked, this, &DeletionConfirmationDialog::updateRememberButtonState);
-        m_ui->buttonBox->button(QDialogButtonBox::Cancel)->setFocus();
-
-        Utils::Gui::resize(this);
-    }
+    DeletionConfirmationDialog(QWidget *parent, int size, const QString &name, bool defaultDeleteFiles);
+    ~DeletionConfirmationDialog();
 
-    ~DeletionConfirmationDialog()
-    {
-        delete m_ui;
-    }
-
-    bool shouldDeleteLocalFiles() const
-    {
-        return m_ui->checkPermDelete->isChecked();
-    }
-
-    static bool askForDeletionConfirmation(QWidget *parent, bool &deleteLocalFiles, const int &size, const QString &name)
-    {
-        DeletionConfirmationDialog dlg(parent, size, name, deleteLocalFiles);
-        if (dlg.exec() == QDialog::Accepted) {
-            deleteLocalFiles = dlg.shouldDeleteLocalFiles();
-            return true;
-        }
-        return false;
-    }
+signals:
+    void accepted(bool isDeleteFileSelected);
```

Comment:
IMO, overriding signals isn't good idea from design perspective.
I would just add an appropriate getter `isDeleteFileSelected()`.

Reply:
>IMO, overriding signals isn't good idea from design perspective.

Just want to be sure, this is actually overloading (`QDialog::accepted()` has no parameter), so is your opinion still the same?

Reply:
>Just want to be sure, this is actually overloading 

I didn't mean exactly "overriding". I don't want to change the "meaning" of an existing signal even by adding another one.
In addition, I am a strict follower of the "classical" Observer, when the object only reports about some event, and all data is provided only through its properties.

>so is your opinion still the same?

Yes.

Reply:
OK, updated.